inference-engine/tools/compile_tool/main.cpp

```diff
@@ -310,6 +342,23 @@ static void processPrecisions(InferenceEngine::CNNNetwork &network,
     }
 }
 
+static void processLayout(InferenceEngine::CNNNetwork &network,
+                          const std::string &inputs_layout, const std::string &outputs_layout) {
+    if (!inputs_layout.empty()) {
+        auto layout = getLayout(inputs_layout);
+        for (auto &&layer : network.getInputsInfo()) {
+            layer.second->setLayout(layout);
```

Comment:
Please check that layout is applicable, eg. it corresponds to the number of dimensions.

Reply:
This check already exists inside the `setLayout` function.

Reply:
No, `setLayout` just checks that `std::string` value can be converted into Layout enumeration. I was talking about layout applicability to tensor. For example, you can't set NC layout for 4D tensor.

Reply:
You need to get number of dimensions from `layout` and call `layer.second->setLayout` only if the number of dimensions matches the `layer.second->getDims().size()`.

Reply:
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/inference_engine/ie_layouts.cpp#L97

Reply:
It throws exception. I suggest just not to call `setLayout` in unsupported case (as it is done for precision). For example, we have networks with 2 inputs 4D and 1D. Passing `-il NHWC` means that we should change layout only for the first input and do nothing with the second one.

Reply:
Done.